### PR TITLE
Add fiscal country column to tables with tax_ids

### PR DESCRIPTION
**Pull request by Vasco-F**
Add support to persist the tax_id origin country for every table with a tax_id column. The field was already being passed to the tax_registration setter but not persisted, so to add support it was only needed to add the getter and setter in the CustomerEntity and Customer interfaces and adjust the some tests to assert that the field is properly set

### CODE REVIEW

File: billy-core-jpa/src/main/java/com/premiumminds/billy/persistence/entities/jpa/JPABusinessEntity.java [hunk 1]
```diff
@@ -164,6 +172,11 @@ public void setFinancialID(String id) {
         this.taxId = id;
     }
 
+    @Override
+    public void setFinancialIDCountry(final String isoCountryCode) {
+        this.taxIdCountry = isoCountryCode;
```
Comment (mabrcosta) [inline billy-core-jpa/src/main/java/com/premiumminds/billy/persistence/entities/jpa/JPABusinessEntity.java:177]: Taking into account the input argument `isoCountryCode` shouldn't the methods, entity field naming and database column a include the reference to iso country code? Or the oposite, and remove the argument naming and update upstream.
As it is it feels like there's contextual knowledge being lost on this level that either is important and should be kept or should not exist to begin with anywhere.

File: billy-core-jpa/src/main/java/com/premiumminds/billy/persistence/entities/jpa/JPACustomerEntity.java [hunk 2]
```diff
@@ -146,6 +154,11 @@ public void setTaxRegistrationNumber(String number) {
         this.taxId = number;
     }
 
+    @Override
+    public void setTaxRegistrationNumberCountry(final String country) {
```
Comment (mabrcosta) [inline billy-core-jpa/src/main/java/com/premiumminds/billy/persistence/entities/jpa/JPACustomerEntity.java:158]: The same solution for the previous comment should also be applied here, whatever the chosen direction will be. Here there is consistency with the input naming, but on the other hand there is inconsistency with the previous place, when both should be consistent.

**Review by mabrcosta — CHANGES_REQUESTED** [review]
Just some naming inconsistencies that should be sorted across the proposed changes.

File: billy-core-jpa/src/main/java/com/premiumminds/billy/persistence/entities/jpa/JPABusinessEntity.java (hunk 1 above)
Comment (Vasco-F) [reply billy-core-jpa/src/main/java/com/premiumminds/billy/persistence/entities/jpa/JPABusinessEntity.java:177]: Shoudn't i keep the same naming schema for the new column as the one adopted for the `taxId` column? That's why i keeped `taxIdCountry`.

However i could use `taxIdISOCountryCode`, while the getter and setter would be `get/setFinancialIdISOCountryCode` to keep them consistent with the existing `get/setFinancialID`. Does that seem adequate??

Comment (mabrcosta) [reply billy-core-jpa/src/main/java/com/premiumminds/billy/persistence/entities/jpa/JPABusinessEntity.java:177]: Sounds good :+1:

**Review by mabrcosta — APPROVED** [review]

File: billy-core/src/main/resources/com/premiumminds/billy/core/i18n/FieldNames.properties [hunk 3]
```diff
@@ -33,6 +33,7 @@ field.application_name=Application Name
 field.version=Version
 field.developer_name=Developer Name
 field.developer_tax_id=Developer Tax Identifier
+field.developer_tax_id_iso_country_code=Developer Tax Identifier ISO Country Code
```
Comment (froque) [inline billy-core/src/main/resources/com/premiumminds/billy/core/i18n/FieldNames.properties:36]: This should be added other languages

```
$ git ls-files \*FieldNames\*properties
billy-core/src/main/resources/com/premiumminds/billy/core/i18n/FieldNames.properties
billy-france/src/main/resources/com/premiumminds/billy/france/i18n/FieldNames_fr.properties
billy-portugal/src/main/resources/com/premiumminds/billy/portugal/i18n/FieldNames_pt.properties
billy-spain/src/main/resources/com/premiumminds/billy/spain/i18n/FieldNames_es.properties
```

**Review by froque — CHANGES_REQUESTED** [review]

**Review by froque — APPROVED** [review]

**Pull request merged**